test/test_common/simulated_time_system_test.cc

```diff
@@ -289,60 +289,57 @@ TEST_P(SimulatedTimeSystemTest, WaitFor) {
   EXPECT_EQ(start_system_time_, time_system_.systemTime());
 
   // Run an event loop in the background to activate timers.
-  std::atomic<bool> done(false);
-  auto thread = Thread::threadFactoryForTest().createThread([this, &done]() {
-    while (!done) {
+  absl::Mutex mutex;
+  bool done(false);
+  auto thread = Thread::threadFactoryForTest().createThread([this, &mutex, &done]() {
+    for (;;) {
+      {
+        absl::MutexLock lock(&mutex);
+        if (done) {
+          return;
+        }
+      }
+
       base_scheduler_.run(Dispatcher::RunType::Block);
     }
   });
-  Thread::MutexBasicLockable mutex;
-  Thread::CondVar condvar;
+
   TimerPtr timer = scheduler_->createTimer(
-      [&condvar, &mutex, &done]() {
-        Thread::LockGuard lock(mutex);
+      [&mutex, &done]() {
+        absl::MutexLock lock(&mutex);
         done = true;
-        condvar.notifyOne();
       },
       dispatcher_);
   timer->enableTimer(std::chrono::seconds(60));
 
-  // Wait 50 simulated seconds of simulated time, which won't be enough to
-  // activate the alarm. We'll get a fast automatic timeout in waitFor because
-  // there are no pending timers.
+  // Wait 1ms. waitFor() does not advance simulated time, so this is just going to verify that
```

Comment:
Wait 1ms of real time.

(just add "of real time" as it might not be obvious looking here).